core/shared/src/main/scala/fs2/internal/CompileScope.scala

```diff
@@ -121,40 +121,41 @@ private[fs2] final class CompileScope[F[_]] private (
      *     or as a result of interrupting this scope. But it should not propagate its own interruption to this scope.
      *
      */
-    def createScopeContext: F[(Option[InterruptContext[F]], Token)] = {
-      val newScopeId = new Token
+    def createScopeContext(newScopeId: Token): F[Option[InterruptContext[F]]] =
       self.interruptible match {
         case None =>
-          F.pure(InterruptContext.unsafeFromInteruptible(interruptible, newScopeId) -> newScopeId)
+          F.pure {
+            interruptible.map(concf =>
+              InterruptContext.unsafeFromConcurrent[F](concf, newScopeId, F.unit))
+          }
 
         case Some(parentICtx) =>
-          F.map(parentICtx.childContext(interruptible, newScopeId))(Some(_) -> newScopeId)
+          F.map(parentICtx.childContext(interruptible, newScopeId))(Some(_))
       }
-    }
-
-    F.flatMap(createScopeContext) {
-      case (iCtx, newScopeId) =>
-        F.flatMap(state.modify { s =>
-          if (!s.open) (s, None)
-          else {
-            val scope = new CompileScope[F](newScopeId, Some(self), iCtx)
-            (s.copy(children = scope +: s.children), Some(scope))
-          }
-        }) {
-          case Some(s) => F.pure(Right(s))
-          case None    =>
-            // This scope is already closed so try to promote the open to an ancestor; this can fail
-            // if the root scope has already been closed, in which case, we can safely throw
-            self.parent match {
-              case Some(parent) =>
-                F.flatMap(self.interruptible.map(_.cancelParent).getOrElse(F.unit)) { _ =>
-                  parent.open(interruptible)
-                }
 
-              case None =>
-                F.pure(Left(new IllegalStateException("cannot re-open root scope")))
-            }
+    val newScopeId = new Token
+    F.flatMap(createScopeContext(newScopeId)) { iCtx =>
```

Comment:
Something I have noticed  is that this gets created, but immediately becomes garbage, if the state is not open. Would it make more sense to put it inside that branch?